eng/tools/GenerateFiles/Directory.Build.targets.in

```diff
@@ -108,7 +109,7 @@
     <KnownCrossgen2Pack Update="Microsoft.NETCore.App.Crossgen2">
       <Crossgen2PackVersion
           Condition=" '%(TargetFramework)' == '${DefaultNetCoreTargetFramework}' ">${MicrosoftNETCoreAppRuntimeVersion}</Crossgen2PackVersion>
-      <Crossgen2RuntimeIdentifiers Condition=" '$(DotNetBuildSourceOnly)' == 'true' ">$(TargetRuntimeIdentifier)</Crossgen2RuntimeIdentifiers>
+      <Crossgen2RuntimeIdentifiers Condition="'$(DotNetBuildOrchestrator)' == 'true'">$(NETCoreSdkRuntimeIdentifier);$(TargetRuntimeIdentifier)</Crossgen2RuntimeIdentifiers>
```

Comment:
OK this is a bit problematic. I was going to override the `NETCoreSdkRuntimeIdentifier` as it controls what the SDK is bringing in: https://github.com/dotnet/sdk/pull/45362

But if I now override that property then we don't restore the host crossgen2 pack anymore. Maybe we need an SDK change to allow a more fine-grained control over the pack RIDs that are being restored? I.e. there's already a property that controls the apphost pack's RID but there isn't one for the other packs.

Reply:
I noticed that you are adding the host rid in https://github.com/dotnet/windowsdesktop/pull/4815/files. Why is that?

Reply:
> OK this is a bit problematic. I was going to override the `NETCoreSdkRuntimeIdentifier` as it controls what the SDK is bringing in: [dotnet/sdk#45362](https://github.com/dotnet/sdk/pull/45362)
> 
> But if I now override that property then we don't restore the host crossgen2 pack anymore. Maybe we need an SDK change to allow a more fine-grained control over the pack RIDs that are being restored? I.e. there's already a property that controls the apphost pack's RID but there isn't one for the other packs: [dotnet/sdk@`4a7aa40`/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets#L157](https://github.com/dotnet/sdk/blob/4a7aa40342ebf9095014de685fcb399c3e497db7/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets#L157)

Yeah, we can't just override `NETCoreSdkRuntimeIdentifier` in dotnet/sdk#45362 as it controls how the SDK selects tools that it knows can run on the host machine. It works fine for Win-x86 because we can run Win-x86 tools on a Win-x64 build machine, but if we tried to do that for linux-arm64 for example, things would break pretty badly (crossgen would just completely fail to execute).

I think we need different/additional controls within the SDK to control which pack RIDs are restored instead of changing the SDK's understanding of its current RID.

> I noticed that you are adding the host rid in https://github.com/dotnet/windowsdesktop/pull/4815/files. Why is that?

In dotnet/windowsdesktop#4815 I don't need to add the host RID as we aren't changing the RID list there. We aren't changing the RID list as we never need to add the "target RID" to the list because we're never building windowsdesktop with a nonportable RID.